obsei/analyzer/ner_analyzer.py

```diff
@@ -100,3 +101,51 @@ def analyze_input(
                     )
                 )
         return analyzer_output
+
+
+class SpacyNERAnalyzer(BaseAnalyzer):
+    _nlp: en.English = PrivateAttr()
+    _max_length: int = PrivateAttr()
+    TYPE: str = "NER"
+    model_name_or_path: str
+    tokenizer_name: Optional[str] = None
+    grouped_entities: Optional[bool] = True
+
+    def __init__(self, **data: Any):
+        super().__init__(**data)
+        self._nlp = spacy.load(
+            self.model_name_or_path,
+            disable=["tagger", "parser", "attribute_ruler", "lemmatizer"],
+        )
+
+    def analyze_input(
+        self,
+        source_response_list: List[TextPayload],
+        analyzer_config: Optional[BaseAnalyzerConfig] = None,
+        **kwargs,
+    ) -> List[TextPayload]:
+        analyzer_output: List[TextPayload] = []
+        texts = [
+            source_response.processed_text for source_response in source_response_list
+        ]
+
+        for doc, source_response in zip(self._nlp.pipe(texts), source_response_list):
```

Comment:
Please use _batchify to process input data in batches. Similar to done in transformer's analyzers

Reply:
I think current batch size is 64 defined in BaseAnalyzerConfig.

Regarding `vars(ent)` comment, I mean this (it will simplify the code)
```
        for doc, source_response in zip(self._nlp.pipe(texts), source_response_list):
            analyzer_output.append(
                TextPayload(
                    processed_text=source_response.processed_text,
                    meta=source_response.meta,
                    segmented_data={
                           "data":  [vars(ent) for ent in doc.ents]
                    },
                    source_name=source_response.source_name,
                )
            )
```

Reply:
I tried vars(ent) ,  but its not possible since ent is of class Span and it does not have any __dict__ method, so keeping it as is